nix/supervisord-cluster/default.nix

```diff
@@ -1,15 +1,21 @@
+let
+  basePortDefault    = 30000;
+  cacheDirDefault    = "${__getEnv "HOME"}/.cache";
+  stateDirDefault    = "state-cluster";
+  profileNameDefault = "default-mary";
+in
 { pkgs
 , lib
 , bech32
-, basePort ? 30000
-, stateDir ? "state-cluster"
-, cacheDir ? "${__getEnv "HOME"}/.cache"
+, basePort              ? basePortDefault
+, stateDir              ? stateDirDefault
+, cacheDir              ? cacheDirDefault
```

Comment:
I see that pattern in nix all the time and I often ask me the question,
if that function is ever called with arguments that override the defaults.
If indeed it is, would it be better to make these arguments just regular/ mandatory arguments.
(or use partial application to specialize the function).
Somehow it seems to be common in nix to wrap every argument in a `Maybe`.
Maybe here is a case for `convention over configuration` and it is better to just decide for good that `state-dir=="state-cluster"`

Reply:
An author of a library may wish to provide a generic and reusable API (because the user does not want to not modify the source code of the library).
But this is not a library. It is just a helper function that is not meant to be reused by third party.

Reply:
Is this `"${__getEnv "HOME"}/.cache"` similar to `unsafePerformIO` ?

Reply:
Yes,  `cacheDir` computation pierces the abstraction, for the sake of providing a convenient default people will use in like 99.9% of time : -)